src/cplus_plugin/gui/ncs_pathway_editor_dialog.py

```diff
@@ -523,11 +575,51 @@ def _on_default_layer_selection_changed(self):
 
     def _on_default_layer_source_selection_changed(self):
         """Event raised when online layer source selection is changed."""
+        self.cbo_default_layer.clear()
+        self.cbo_default_layer.addItem("")
+
         source = self.cbo_default_layer_source.currentText()
         if source.lower() == LayerSource.CPLUS.value.lower():
             items = sorted([p["name"] for p in self.cplus_list])
+            self.cbo_default_layer.addItems(items)
+
         elif source == LayerSource.NATUREBASE.value:
-            items = sorted([p["name"] for p in self.naturebase_list])
-        self.cbo_default_layer.clear()
-        self.cbo_default_layer.addItem("")
-        self.cbo_default_layer.addItems(items)
+            for group_name, options in self.naturebase_pathway_type_groups.items():
+                self.add_online_group(
+                    group_name, options, self.cbo_default_layer.model()
+                )
+
+    def add_online_group(self, group_name, options, model):
+        # Group header
+        group_item = QtGui.QStandardItem(group_name)
+        group_item.setEnabled(False)
+        group_item.setSelectable(False)
+        group_item.setData("group", QtCore.Qt.ItemDataRole.UserRole)
+        group_item.setForeground(QtCore.Qt.GlobalColor.darkGray)
+
+        model.appendRow(group_item)
+
+        carbon_options = {}
+
+        # Group options
+        for option in options:
+            carbon_value = -1
+            impact = self._get_layer_carbon_info(option)
+            if impact:
+                carbon_value = impact.get(MEAN_VALUE_ATTRIBUTE, -1)
+            carbon_options[option] = carbon_value
+
+        sorted_carbon_list = sorted(
+            carbon_options.items(), key=lambda item: item[1], reverse=True
+        )
+        sorted_carbon_options = dict(sorted_carbon_list)
+
+        for option in sorted_carbon_options:
+            impact = self._get_layer_carbon_info(option)
+            label = f"  {option}"
+            carbon_value = impact.get(MEAN_VALUE_ATTRIBUTE)
+            if carbon_value:
```

Comment:
`if carbon_value:` will be False if `carbon_value = 0`. Rather use `if carbon_value is not None:`, unless we need to skip 0.